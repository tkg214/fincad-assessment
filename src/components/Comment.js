import React from 'react';
import PropTypes from 'prop-types';

const Comment = props => {
  return (
    <div>
      <h5 data-test="indexCount">
        {`Comment ${props.index + 1} of ${props.count}`}
      </h5>
      <p data-test="nameEmail">
        {`by ${props.name} (${props.email})`}
      </p>
      <p data-test="body">
        {`${props.body}`}
      </p>
    </div>
  );
};

Comment.propTypes = {
  index: PropTypes.number.isRequired,
  count: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired
};

export default Comment;
