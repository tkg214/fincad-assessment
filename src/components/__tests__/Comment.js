import React from 'react';
import { render, mount } from 'enzyme';

import Comment from '../Comment';

test('the comment component renders with defaults', () => {
  const wrapper = renderComment();
  expect(wrapper).toMatchSnapshot();
});

test('the component renders with props passed in', () => {
  const index = 0;
  const count = 1;
  const name = 'Test Name';
  const email = 'test@test.com';
  const body = 'Test Body';
  const wrapper = mountComment({ index, count, name, email, body });
  expect(wrapper.find(sel('indexCount')).text()).toBe(
    `Comment ${index + 1} of ${count}`
  );
  expect(wrapper.find(sel('nameEmail')).text()).toBe(`by ${name} (${email})`);
  expect(wrapper.find(sel('body')).text()).toBe(body);
});

function getCommentProps(props = {}) {
  return {
    index: 0,
    count: 0,
    name: '',
    email: '',
    body: '',
    ...props
  };
}

function renderComment() {
  const commentProps = getCommentProps();
  return render(<Comment {...commentProps} />);
}

function mountComment(props) {
  const commentProps = getCommentProps(props);
  return mount(<Comment {...commentProps} />);
}

function sel(id) {
  return `[data-test="${id}"]`;
}
