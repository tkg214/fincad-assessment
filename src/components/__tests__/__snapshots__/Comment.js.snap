// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the comment component renders with defaults 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "attribs": Object {},
        "children": Array [
          Object {
            "attribs": Object {
              "data-test": "indexCount",
            },
            "children": Array [
              Object {
                "data": "Comment 1 of 0",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "h5",
            "next": Object {
              "attribs": Object {
                "data-test": "nameEmail",
              },
              "children": Array [
                Object {
                  "data": "by  ()",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "p",
              "next": Object {
                "attribs": Object {
                  "data-test": "body",
                },
                "children": Array [],
                "name": "p",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
              },
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
          },
          Object {
            "attribs": Object {
              "data-test": "nameEmail",
            },
            "children": Array [
              Object {
                "data": "by  ()",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "p",
            "next": Object {
              "attribs": Object {
                "data-test": "body",
              },
              "children": Array [],
              "name": "p",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
            },
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "data-test": "indexCount",
              },
              "children": Array [
                Object {
                  "data": "Comment 1 of 0",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "h5",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
            },
            "type": "tag",
          },
          Object {
            "attribs": Object {
              "data-test": "body",
            },
            "children": Array [],
            "name": "p",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "data-test": "nameEmail",
              },
              "children": Array [
                Object {
                  "data": "by  ()",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "p",
              "next": [Circular],
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "data-test": "indexCount",
                },
                "children": Array [
                  Object {
                    "data": "Comment 1 of 0",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "h5",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
              },
              "type": "tag",
            },
            "type": "tag",
          },
        ],
        "name": "div",
        "next": null,
        "parent": null,
        "prev": null,
        "root": [Circular],
        "type": "tag",
      },
    ],
    "name": "root",
    "next": null,
    "parent": null,
    "prev": null,
    "type": "root",
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xmlMode": false,
  },
}
`;
